src/app/admin/components/filter-search/filter-search.component.ts

```diff
@@ -77,34 +77,51 @@ export class FilterSearchComponent implements OnInit {
 
     //check for params in the query and add them to the filter dropdown bars
     const qParams = this.route.parent.snapshot.queryParams;
+
+    /**
+     * TODO: refactor this to send a complete search query
+     * at the first request instead of sending a request
+     * every time we emit an event on the toggle filter functions.
+     *
+     * this COULD be a cause of the client sending bad data back
+     * to the user, so just be aware.
+     */
+
     //if there are topics in the query, toggle them in the filter dropdown
-    if (qParams.topics) {
-      //more than one topic
+    if (qParams.topics && qParams.topics.length > 0) {
+      const topics = [];
       if (Array.isArray(qParams.topics)) {
```

Comment:
Instead of having to do this type checking can we enforce the type sent to the component?

Reply:
removed